core/include/kernel/rpc_io_i2c.h

```diff
@@ -0,0 +1,32 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2020 Foundries Ltd <jorge@foundries.io>
+ */
+
+#ifndef __RPC_IO_I2C_H
+#define __RPC_IO_I2C_H
+
+#include <tee_api_types.h>
+
+enum rpc_i2c_mode {
+	RPC_I2C_MODE_WRITE = 0,
+	RPC_I2C_MODE_READ = 1,
```

Comment:
This addresses master read/write transfer. Do you plan to support combined transfer (aka register/memory read/write, where a master emits write to specify target address read/written in slave, followed by effective read/write transfer? Or maybe caller shall issue several of these nominal i2c rpc transfers to do so.

Reply:
no not really, this was supposed to be the simplest possible way to communicate with an i2c device like an HSM. it is not more ambitious than that.

Reply:
I guess this is ok then.